waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -26,10 +23,8 @@ import Wasp.Cli.Common (WaspProjectDir)
 import qualified Wasp.Message as Msg
 import qualified Wasp.Util.Terminal as Term
 
--- It receives all of the arguments that were passed to the `wasp new` command.
-createNewProject :: Arguments -> Command ()
-createNewProject args = do
-  newProjectArgs <- parseNewProjectArgs args & either throwProjectCreationError return
+createNewProject :: NewArgs -> Command ()
+createNewProject newProjectArgs = do
```

Comment:
I was looking at `NewArgs` here, wondering where is the right place to define them, and ended up thinking generally about how we define Commands in our CLI.

So a bit of general thinking! Let's say we want to add a new command to Wasp CLI. We would need to define following stuff:
1. Command parser (for optparse) -> ok, this might be trivial, so it is just `pure Smth`, but it can be a proper parser also.
2. Command Call.
3. Command args type. Sometimes there are none, sometimes it is just list of Strings or something simple, but often it is a custom type (record).
4. Command logic that actually executes the command (in Command monad).
5. Name and usage/description for a command.

Some relations: Args (3) are used by Call (2) and command logic (4). Parser (1) generates Call (2), so it also transitively uses Args (3). Parser (1) and name and usage (5) are combined together to form the final entry for optparse, to represent the command.

Ok, that was a bit of analysis!

The question is, where do we put all these parts, in the codebase? Right now Parser (1) is sometimes next to logic (4) or is directly or is directly in optparse definition of the whole CLI. Name and usage (5) is sometimes together with Parser (1), sometimes it is directly in the optparse definition in the CLI. Args (3) are right now defined next to Command Call (2).

One thing we can improve for sure here is ensure things are consistent. Main inconsistency right now is that name and usage are sometimes in specific command Parser, but sometimes they are directly in the top level Parser -> I would make sure they are always in the top level Parser.

Next, it would be interesting to explore if we could collocate things more.
What if every command would have all its parts grouped?
So we would have all parts, from 1 to 5, in the same module, be it in multiple sub-modules or in one sub-module.
So for example, for command New, we would have its logic in one file, parser in file next to it, args maybe in the same file as logic, name and usage possibly in the same file as parser like you were doing now. I am not sure only about Command Call -> that one maybe we could leave outside, where it is now. Yeah, I think so.

I am not 100% sure about this, it would require a bit of playing and seeing what it looks, but I think something in this direction might be interesting. If you feel it is too much for this PR, we can possibly leave this as the very last thing to play with, or even for the future PR. One thing I would address is just making sure that name and usage are always in the same place.

Reply:
@modotte I am curious what you think after reading this -> if you have a good idea how to improve some of the stuff here and want to try out, that could be interesting.
If you don't and any direction you pick looks like more trouble than it is worth, then I am completely fine with leaving it as it is for now.

Reply:
I don't really have a clear idea of how to do this, unfortunately. Maybe we should leave this one in the future/separate issue/for a different PR? 

Although, I'd really like the idea of bringing the relevant things closer together for cohesiveness. That certainly would make it easier if someone else wants to add/remove/update the commands in the future.

Reply:
Yup, the collocation would be nice! I am ok with leaving this as it is for now then, unless you want to try something out. So I will leave it to you, feel free to resolve this comment if you think it would be better to leave this for the future (which makes sense), or keep it open if you have some ideas / approaches and we can work on that.